Readme-edit

sghrm opened the pull request:
# WHAT
ER図の作成とReadmeへのDB設計を記述致しました。

※修正・追加コミットに伴いER図も修正しました。
https://gyazo.com/157debbb1c433cb5f59d0fde36ae3a9a

ご確認お願い致します。
# WHY
今後の設計に必要なため

sghrm committed:
Readme-edit

hzak2xx commented on README.md:
```diff
@@ -22,3 +22,151 @@ Things you may want to cover:
+- belongs_to :item
+- belongs_to :user
+
+## selesテーブル
```
これは何ですか？スペルが違う気がします

sghrm replied:
スペル間違いでした。修正しました

hzak2xx commented on README.md:
```diff
@@ -22,3 +22,151 @@ Things you may want to cover:
+
+|Column|Type|Options|
+|------|----|-------|
+|user_id|references|null: false, foreign_key: true|
```
commentのtextは必要ないですか？

sghrm replied:
ER図に記載していながら抜けてました。追記します。

hzak2xx commented on README.md:
```diff
@@ -22,3 +22,151 @@ Things you may want to cover:
+### Association
+- belongs_to :user
+
+## pointsテーブル
```
具体的に何を指しているかわからないので名前変更してください

sghrm replied:
フリーマーケットで使えるポイントの意で
freemarket_pointsに変更しました。

hzak2xx commented on README.md:
```diff
@@ -22,3 +22,151 @@ Things you may want to cover:
+|Column|Type|Options|
+|------|----|-------|
+|notification_title|string|null: false|
+|notification_content|text|null: false|
```
ただ単にbelongs_to useだけだと足りない気がするので最低でも通知元、通知先のuserは欲しいです。

hzak2xx replied:
このタイプのnotificationだと、ポリモーフィック関連を使ってもいいと思うので検討してみてください

sghrm replied:
ポリモーフィック関連を使うようにします。

hzak2xx commented on README.md:
```diff
@@ -22,3 +22,151 @@ Things you may want to cover:
+|Column|Type|Options|
+|------|----|-------|
+|todo_title|string|null: false|
+|todo_content|text|null: false|
```
`todo_`は不要です

sghrm replied:
todo_　を抜きました。

sghrm committed:
Readme-modify

# WHAT
・selesテーブルのスペル修正
・comementテーブルcontentカラム追加
・pointsテーブルのテーブル名修正
・notificationテーブルのポリモーフィック関連導入
・todoテーブルのカラム名修正

miyagit commented on README.md:
```diff
@@ -22,3 +22,155 @@ Things you may want to cover:
+|family_name_kana|string|null: false, unique: true|
+|first_name_kana|string|null: false, unique: true|
+|nickname|string|null: false, unique: true|
+|mail|string|null: false|
```
mail被ってますね。

sghrm replied:
ご指摘ありがとうございます！片方削除します。

miyagit commented on README.md:
```diff
@@ -22,3 +22,155 @@ Things you may want to cover:
+
+
+### Association
+- has_many :sales
```
基本的にhas_manyには、dependent: :destroyつけた方が良いですね。

sghrm replied:
has_manyに　dependent: :destroy付けました。

miyagit commented on README.md:
```diff
@@ -22,3 +22,155 @@ Things you may want to cover:
+|address|text|null: false|
+|mail|string|null: false|
+|telephone|string|null: false|
+|birthday|string|null: false|
```
こちらdate方の方が良さそうですね。

sghrm replied:
dateにしました！

miyagit commented on README.md:
```diff
@@ -22,3 +22,155 @@ Things you may want to cover:
+
+|Column|Type|Options|
+|------|----|-------|
+|comment_content|text|null: false|
```
commentsテーブルなんで、comment_はなくても大丈夫そうですね。

sghrm replied:
comment_　無くしました。

miyagit commented on README.md:
```diff
@@ -22,3 +22,155 @@ Things you may want to cover:
+
+|Column|Type|Options|
+|------|----|-------|
+|sale|string|null: false|
```
saleとは主に何が入るんですか？

sghrm replied:
salesテーブルは売上のテーブルで
saleは売上数値を入れるように考えています。

miyagit reviewed (commented):
見ていて思ったことが、
基本的にitem_、comment_は必要ないですね。テーブルに書いてあるん。

belongs_to :userとしてるのに外部キーがないのがありました。

あとは型がおかしかったりですね。

上の点に当てはまるものは全て見直して修正してください

sghrm committed:
Readme-modify

・指摘点に対しての修正

ShuheiTsuji commented on README.md:
```diff
@@ -22,3 +22,177 @@ Things you may want to cover:
+|shipping_date|date|null: false|
+|shipping_area|string|null: false|
+|shipping_fee|string|null: false|
+
```
誰が出品したとか管理できるようになってますでしょうか？ 🤔

sghrm replied:
出品者と紐づくように修正します

ShuheiTsuji commented on README.md:
```diff
@@ -22,3 +22,177 @@ Things you may want to cover:
+### Association
+- has_many :comments, dependent: :destroy
+- has_many :likes, dependent: :destroy
+- belongs_to :category
```
category は、このテーブル関係で良いですか！
中間テーブルの方が良いかなと思うのですが。任せます。

sghrm replied:
中間テーブルに致します。

ShuheiTsuji commented on README.md:
```diff
@@ -22,3 +22,177 @@ Things you may want to cover:
+
+|Column|Type|Options|
+|------|----|-------|
+|user_id|references|null: false, foreign_key: true|
```
これって、今、どういった状況とかは管理しなくて良いのでしょうか 🤔

ShuheiTsuji replied:
sells とか sales とかテーブルで分けているのはそういうことですかね。

sghrm replied:
transactionテーブルを作り、status管理するようにします

sghrm committed:
DB-edit

指摘いただいた点の修正を実施しました。
実装する予定のないテーブルを削除しました。

kuroko918 commented on README.md:
```diff
@@ -36,47 +36,60 @@ Things you may want to cover:
 |first_name_kana|string|null: false, unique: true|
```
氏名がunique: trueなのはまずいかなと思います。
同姓同名がいると思うので。

sghrm replied:
修正致します！

kuroko918 commented on README.md:
```diff
@@ -36,47 +36,60 @@ Things you may want to cover:
 |nickname|string|null: false, unique: true|
 |mail|string|null: false|
-|address|text|null: false|
 |telephone|string|null: false|
```
ここはunique: tureでもいいかと思います。

sghrm replied:
追記します！

kuroko918 commented on README.md:
```diff
@@ -36,47 +36,60 @@ Things you may want to cover:
-- belongs_to :category
+- has_many :transactions, dependent: :destroy
+- has_many :images, dependent: :destroy
+- has_many :categorys,through,category_croups
```
`through: :category_croups`に書き換えておいてください！

sghrm replied:
スペルミス含め、修正します！

kuroko918 commented on README.md:
```diff
@@ -102,83 +115,29 @@ Things you may want to cover:
-|user_id|references|null: false, foreign_key: true|
-|item_id|references|null: false, foreign_key: true|
+|name|string|index: true,null: false, unique: true|
+|parent_id|integer|---|
```
parent_idとはなんでしょう？

sghrm replied:
カテゴリの階層を表すために、
親カテゴリがある場合にそのIDを取得するカラムです。

kuroko918 commented on README.md:
```diff
@@ -102,83 +115,29 @@ Things you may want to cover:
-### Association
-- belongs_to :user
+- has_many :items, through: :category_groups
+- has_many :category_groups
```
外部キーがあるのに、has_manyはおかしいです。
belongs_toではないでしょうか？

sghrm replied:
belongs_toです。修正します。

kuroko918 commented on README.md:
```diff
@@ -187,12 +146,16 @@ Things you may want to cover:
-- has_many :items, dependent: :destroy
+- belongs_to :item
+- belongs_to :user
+- belongs_to :buyer, class_name: "User"
```
ここは:sellerのアソシエーションも必要かと思います！

sghrm replied:
sellerアソシエーションを追加しました！

kuroko918 commented on README.md:
```diff
@@ -36,47 +36,60 @@ Things you may want to cover:
-- has_many :purchases, dependent: :destroy
-- has_many :sells, dependent: :destroy
+- has_many :transactions, dependent: :destroy
+- has_many :buyer, class_name: "Transaction"
```
foreign_key: :buyer_idも必要かと思います。

あと、ここは:sellerのアソシエーションも必要かと思います！

sghrm replied:
sellerのアソシエーションを追加しました！
foreign_key: :buyer_id
foreign_key: :seller_id
追加しました。

kuroko918 reviewed (commented):
いろいろと不備があったので、修正お願いします。
取引に関することは、こちらの記事を参考にしてください。(http://www.coma-tech.com/archives/223/)

sghrm committed:
Readme-edit

指摘点に対しての修正
seller_id追加

kuroko918 commented on README.md:
```diff
@@ -22,3 +22,143 @@ Things you may want to cover:
+- has_many :transactions, dependent: :destroy
+- has_many :images, dependent: :destroy
+- has_many :categorys,through: :category_groups
+- belongs_to :brand
```
ここにuserとのアソシエーションも必要かと思います。

sghrm replied:
追記しました。

kuroko918 commented on README.md:
```diff
@@ -22,3 +22,143 @@ Things you may want to cover:
+- has_many :items, dependent: :destroy
+- has_many :comments, dependent: :destroy
+- has_many :likes, dependent: :destroy
+- has_many :transactions, dependent: :destroy
```
これは必要ないかと思います。

sghrm replied:
削除しました！

kuroko918 commented on README.md:
```diff
@@ -22,3 +22,143 @@ Things you may want to cover:
+
+### Association
+- has_one :credit_cards, dependent: :destroy
+- has_many :items, dependent: :destroy
```
itemとのアソシエーションはsellerとbuyerに分けて記述しなくてはいけません。

sghrm replied:
記事の内容を理解しきれていませんでした。
再度読み直して修正しました。

kuroko918 commented on README.md:
```diff
@@ -22,3 +22,143 @@ Things you may want to cover:
 * Deployment instructions
 
 * ...
+
```
この行から上はいらないと思うので消しておいてください！

sghrm replied:
削除しました！

kuroko918 reviewed (commented):
コメントしました、修正お願いします。
もう一度、お渡しした記事をよく読んで、コメント部分のアソシエーションを修正して下さい!

sghrm committed:
Readme-edit

修正の反映

kuroko918 commented on README.md:
```diff
@@ -89,9 +66,12 @@ Things you may want to cover:
 - has_many :images, dependent: :destroy
 - has_many :categorys,through: :category_groups
 - belongs_to :brand
+- belongs_to :user
```
これは要らないかと思います。
```
- has_many :sellers, through: :transactions
- has_many :buyers, through: :transactions
```
これで必要十分です。あとuserとitemは多対多の関係にあります。

sghrm replied:
belongs_to :user　は削除しました。
多対多を理解できていませんでした。
itemsテーブルの
user_idも不要になるかと思いますので削除します。

sghrm committed:
Readme-edit

Itemsテーブルの修正
belongs_to userの削除
user_idカラムの削除

kuroko918 commented:
いいかと思います！
LGTMです！
